modules/loader/src/main/scala/com/snowplowanalytics/snowplow/postgres/source/LocalEnv.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.postgres.source
+
+import cats.implicits._
+import cats.effect.{ContextShift, Blocker, Resource, Timer, ConcurrentEffect}
+
+import fs2.{Pipe, Stream}
+
+import blobstore.fs.FileStore
+import blobstore.Store
+
+import com.snowplowanalytics.snowplow.postgres.streaming.SinkPipe
+import com.snowplowanalytics.snowplow.postgres.config.LoaderConfig.{Source}
+import com.snowplowanalytics.snowplow.postgres.streaming.data.BadData
+
+object LocalEnv {
+
+  def create[F[_]: ConcurrentEffect : ContextShift : Timer](blocker: Blocker, config: Source.LocalFS): Resource[F, Environment[F, String]] = {
+    Resource.eval {
```

Comment:
I think we could use a weaker abstraction at construction-site, `F[Environment[F, String]]` and lift it to `Resource` at call-site.

Reply:
`Resource.eval` is not used in the `KinesisEnv` therefore we will not able to wrap all of the environment `create` method returns with `Resource.eval` at the call-site. Also, we can individually wrap result of `LocalEnv` and `PubsubEnv` create methods at the call side but I think calling them in the create method is better because it creates unified interface and all environment create methods are returning Resource.